File: src/learning_time_py/learning_time_py/duration_demo.py
import rclpy
from rclpy.node import Node
from rclpy.duration import Duration

class DurationDemo(Node):
    def __init__(self):
        super().__init__("duration_demo_py")
        du1 = Duration(seconds = 2, nanoseconds = 500000000)
        self.get_logger().info(
            f"ns = {du1.nanoseconds}"
        )

def main():
    rclpy.init()
    node = DurationDemo()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/learning_time_cpp/src/duration_demo.cpp
#include "rclcpp/rclcpp.hpp"

class DurationDemo : public rclcpp::Node {
public:
    DurationDemo() : Node("duration_demo_cpp") {
        rclcpp::Duration du1(2, 500000000);
        RCLCPP_INFO(
            this->get_logger(),
            "s = %.2f, ns = %ld",
            du1.seconds(), du1.nanoseconds()
        );
    }
private:
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<DurationDemo>();
    rclcpp::shutdown();

    return 0;
}